frontend/src/components/G6Graph.svelte

```diff
@@ -36,127 +36,245 @@
+    graph.addItem("node", ev, false, false);
+    for (const edge of ev.edges) {
+      graph.addItem("edge", {
+        source: ev.id,
+        target: edge.target,
+        label: edge.type,
+      }); // the type of link is connected to the label of the edge here.
+      edgesToBack(ev); // put all edges attached to the node behond the nodes (needed when using groupByTypes: false);
+    }
 
-        timeBarData.push({
-            date: ev.date,
-            value: "1",
-        });
-    };
+    timeBarData.push({
+      date: ev.date,
+      value: "1",
+    });
+  };
 
-    export const updateTimeBar = (timeBarEnabled: boolean) => {
-        graph.removePlugin(graph.get("plugins")[0]);
-        if (!timeBarEnabled) {
-            //TO-DO Reset the graph if wanted later
-        } else {
-            graph!.addPlugin(
-                new G6.TimeBar({
-                    className: "g6TimeBar",
-                    x: 0,
-                    y: 0,
-                    width: 500,
-                    height: 110,
-                    padding: 10,
-                    type: "trend",
-                    changeData: false,
-                    trend: {
-                        data: timeBarData,
-                        smooth: true,
-                    },
-                    tick: {
-                        tickLabelFormatter: (timeBarData: any) => {
-                            return "";
-                        },
-
-                        tickLineStyle: {
-                            fill: "#f28c18",
-                        },
-                    },
-                    slider: {
-                        backgroundStyle: {
-                            fill: "#131616",
-                        },
-                        foregroundStyle: {
-                            fill: "#ffffff",
-                        },
-                        handlerStyle: {
-                            style: {
-                                fill: "#f28c18",
-                                stroke: "#f28c18",
-                            },
-                        },
-                    },
-                    controllerCfg: {
-                        fill: "#131616",
-                        stroke: "#131616",
-                        timePointControllerText: " Point",
-                        timeRangeControllerText: " Point",
-                    },
-                })
-            );
-        }
-    };
+  /**
+   * Helper method that will rearrange the order of items on the z-index (edges behind the nodes)
+   * To avoid iterating through the whole graph and update, we use this helper method inside the push method.
+   * This allows to only manipulate the nodes newly pushed into the graph.
+   * @param event containing a node and its edges.
+   */
+  const edgesToBack = (event) => {
+    const node = graph.findById(event.id);
+    if (node instanceof Node) {
+      const edges = node.getEdges();
+      edges.forEach((edge) => {
```

Comment:
A normal for loop would be more readable

Reply:
Quite subjective, I think the for.Each is more readable but I can change it to a normal loop if you wish so.

Reply:
Fixed for the next commit (waiting for answers to questions before pushing to have one commit addressing all feedback)